numpy/ma/core.py

```diff
@@ -5107,8 +5107,12 @@ def round(self, decimals=0, out=None):
 
         """
         result = self._data.round(decimals=decimals, out=out).view(type(self))
-        result._mask = self._mask
-        result._update_from(self)
+        if result.ndim:
```

Comment:
It would be more informative to write `if result.ndim > 0:`, as that exactly expresses the desired condition.